### ISSUE TITLE
conditionals in eap.erb template break startup

### ISSUE THREAD

**Issue by decibelhertz** [djjudas21/puppet-freeradius#94]
Per https://github.com/djjudas21/puppet-freeradius/blob/3.5.0/templates/eap.erb#L126-L159 , a closing curly bracket can be left dangling is `$eap_gtc` is *true* (default) and `$gtc_challenge` is *undef* (default).

Here is the output I see when using nothing but the defaults... `egrep -v '\s*#' /etc/raddb/mods-enabled/eap | grep -v '^$'` (lines 11-12 are questionable):
```
eap {
  default_eap_type = md5
  timer_expire     = 60
  ignore_unknown_eap_types = no
  cisco_accounting_username_bug = no
  max_sessions = ${max_requests}
  md5 {
  }
  leap {
  }
    auth_type = PAP
  }
  tls-config tls-common {
    private_key_file = ${certdir}/server.pem
    certificate_file = ${certdir}/server.pem
    ca_file = ${certdir}/ca.pem
    dh_file = ${certdir}/dh
    ca_path = ${cadir}
    cipher_list = "DEFAULT"
    ecdh_curve = "prime256v1"
    cache {
      enable = yes
      max_entries = 255
    }
    verify {
    }
    ocsp {
      enable = no
      override_cert_url = yes
      url = "http://127.0.0.1/ocsp/"
    }
  }
  tls {
    tls = tls-common
  }
  ttls {
    tls = tls-common
    default_eap_type = md5
    copy_request_to_tunnel = no
    use_tunneled_reply = no
    virtual_server = "inner-tunnel"
  }
  peap {
    tls = tls-common
    default_eap_type = mschapv2
    copy_request_to_tunnel = no
    use_tunneled_reply = no
    virtual_server = "inner-tunnel"
}
  mschapv2 {
  }
}
```
I suspect you want to move the `gtc {` clause to above https://github.com/djjudas21/puppet-freeradius/blob/3.5.0/templates/eap.erb#L138 , but will leave that for you to decide.

Error I am seeing from `radiusd -X`:
```
/etc/raddb/mods-enabled/eap[604]: Too many closing braces
Errors reading or parsing /etc/raddb/radiusd.conf
```

Cheers.

**Comment by djjudas21** [comment]
Thanks for the report. I'll look into this later today.

**Comment by djjudas21** [comment]
Alright, I've fixed this by moving the `gtc {` clause. This slipped through testing because at our site, gtc is explicitly disabled. Can you let me know if this works for you?

**Comment by decibelhertz** [comment]
That worked:
```Notice: /Stage[main]/Profile::Radius/Freeradius::Module::Eap[eap]/Freeradius::Module[eap]/File[/etc/raddb/mods-available/eap]/content: 
--- /etc/raddb/mods-available/eap	2018-01-08 14:51:58.196129217 +0000
+++ /tmp/puppet-file20180111-19206-pm12qr	2018-01-11 14:37:00.455618965 +0000
@@ -97,9 +97,6 @@
   #  for anyone to see.
   #
   gtc {
-    #  The default challenge, which many clients
-    #  ignore..
-    challenge = "super_secret"
 
     #  The plain-text response which comes back
     #  is put into a User-Password attribute,

Notice: /Stage[main]/Profile::Radius/Freeradius::Module::Eap[eap]/Freeradius::Module[eap]/File[/etc/raddb/mods-available/eap]/content: content changed '{md5}79b67764c458693bb02dc99f0fbef563' to '{md5}153a0184'
Notice: Applied catalog in 6.82 seconds
```
... but it revealed another issue.  Changing the content of the module doesn't restart the service.  I restarted it manually and all is well on the original problem.

As for the service problem ... looking at https://github.com/djjudas21/puppet-freeradius/blob/master/manifests/module.pp#L28-L36 ... I think you need a `notify  => Service[$freeradius::fr_service]` clause on that file resource.  Right now, you only restart the service when setting up the symlink.

Alternatively, putting:
```
Freeradius::Module {
  notify => Service[$freeradius::fr_service]
}
```
... in `init.pp` may be a better idea.

**Comment by djjudas21** [comment]
Thanks for reporting that, too. I think this got missed because we used to deploy the module files directly into `mods-enabled` and it was a later change that deployed them into `mods-available` and created the symlink. I've added the
```puppet
Freeradius::Module {
  notify => Service[$freeradius::fr_service]
}
```
block as you suggested. For now I haven't removed the other `notify` but if this approach works, I can probably implement it elsewhere in the module and make things a bit tidier.

**Comment by decibelhertz** [comment]
I upgraded to that commit, made a garbage line in line 2 of `/etc/raddb/mods-available/eap` and Puppet did _not_ work as hoped (same result -- not restarting the service).  I suspect you need to put the new clause *before* all the calls to the defined type, ie above https://github.com/djjudas21/puppet-freeradius/blob/master/manifests/init.pp#L88-L119 . I would put it before any resource (eg https://github.com/djjudas21/puppet-freeradius/blob/master/manifests/init.pp#L36 )

**Comment by djjudas21** [comment]
I will have another look today

**Comment by djjudas21** [comment]
Alright. c183aa9 fixes this

**Comment by decibelhertz** [comment]
Hm.  That didn't do it either.  I'm willing to figure out why, but will need some time.

```
# echo '#foo' >> /etc/raddb/mods-available/eap 
# puppet agent -t 
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for stsmon0.ucsd.edu
Info: Applying configuration version '1c22f1a5e7f5aaed77ac5918eec237ca7770decb'
Notice: /Stage[main]/Profile::Radius/Freeradius::Module::Eap[eap]/Freeradius::Module[eap]/File[/etc/raddb/mods-available/eap]/content: 
--- /etc/raddb/mods-available/eap	2018-01-16 17:56:57.708333589 +0000
+++ /tmp/puppet-file20180116-14730-1giwwqe	2018-01-16 17:57:22.587441031 +0000
@@ -603,4 +603,3 @@
 #    virtual_server = inner-tunnel
 #  }
 }
-#foo

Notice: /Stage[main]/Profile::Radius/Freeradius::Module::Eap[eap]/Freeradius::Module[eap]/File[/etc/raddb/mods-available/eap]/content: 

Notice: /Stage[main]/Profile::Radius/Freeradius::Module::Eap[eap]/Freeradius::Module[eap]/File[/etc/raddb/mods-available/eap]/content: content changed '{md5}6301a6817d65145a8ec9e889fe5e2ab7' to '{md5}153a01845d13df679faea91c829f4254'
Notice: Applied catalog in 6.59 seconds
```
